lib/ddtrace/contrib/mysql2/patcher.rb

```diff
@@ -20,7 +21,15 @@ def patch
         end
 
         def patch_mysql2_client
-          ::Mysql2::Client.include(Instrumentation)
+          # ::Mysql2::Client.include(Instrumentation::Client)
+
+          Datadog::Instrumentation::Hook['Mysql2::Client#query'].add do
+            append do |stack, env|
+              Mysql2::Instrumentation::Client.query(env) do |trace_env|
+                stack.call(trace_env)
+              end
+            end
+          end.install
```

Comment:
Is it common to use `Hook` without performing `install` immediately afterwards? E.g. would it make sense to make this the default behavior?

Reply:
It has been the case in the past, adding multiple hooks on the same hookpoint in different locations, then installing in another part of the lifecycle.

Reply:
I think this would be largely solved with the upcoming tracer instrumentation API, which sits atop of Hook.